### ISSUE TITLE
System Resources/System Services Informations

### ISSUE THREAD

**Issue by fabianoflorentino** [kxr/ocpcr#5]
I have some trouble when tasks get node info, you can help me?

![image](https://user-images.githubusercontent.com/13373667/111803076-7646c000-88ad-11eb-9382-4a4390e47aa8.png)

**Comment by kxr** [comment]
@fabianoflorentino Sure. Once the playbook finishes, you will find a hidden directory in the playbook directory called `.localtmphc` with a bunch of files. One of those files should be `hostvars.json`. Is it possible to share that file here?

**Comment by fabianoflorentino** [comment]
Hi Khizer,

Attach file.

Thanks for your attention :) 

[hostvars.json.zip](https://github.com/kxr/ocpcr/files/6175982/hostvars.json.zip)

**Comment by fabianoflorentino** [comment]
> Hi Khizer,
> 
> Attach file.
> 
> Thanks for your attention :)
> 
> [hostvars.json.zip](https://github.com/kxr/ocpcr/files/6175982/hostvars.json.zip)

Hi Khizer,

did you receive the file?

**Comment by kxr** [comment]
@fabianoflorentino Yes I did receive the file, but unfortunately I got very busy this week. I will check it asap.

**Comment by kxr** [comment]
@fabianoflorentino Sorry for the long delay. I checked the file that you attached. The problem seems to be that Ansible for some reason is missing the core/default hostvars that we are relying on.

So before I explain more I will like to give you a short summary of how this playbook works. The first three blocks in `ocpcr/cluster-report.yaml` collect information from nodes/cluster.

Then, in the fourth block "# Process Cluster Info Locally", the ansible hostvars are [dumped](https://github.com/kxr/ocpcr/blob/master/tasks/local/dump_hostvars.yaml). This is what I requested and you shared. Next values from the dumped hostvars.json file are used to populate the jinja2 templates that make up final the html report.

For example, see [here](https://github.com/kxr/ocpcr/blob/8e072e6ec25b7cd82f131b0dbbc53ea603a8e9fb/report/includes/system_resources.j2#L27) we pickup the `ansible_architecture` value from hostvars.json and populate it in the template. This seems to be working fine, the variable is present in the hostvars.json dump and it is populated in the screenshot that you shared (**Arch** column).

To my surprise, some hostvars are missing for example [ansible_processor_vcpus](https://github.com/kxr/ocpcr/blob/8e072e6ec25b7cd82f131b0dbbc53ea603a8e9fb/report/includes/system_resources.j2#L34), which is used to populate the **vCPU** column. I am not sure why these variables are missing.

May be you are using a newer version of ansible and things have changed. Can you tell me what version of ansible you used to run this playbook. And if possible can you try this playbook with ansible version 2.6?

**Comment by kxr** [comment]
Also, try the following from the bastion host (pick any one node):

```
ansible -i /path/to/inventory <node-name> -m setup | grep "ansible_processor_vcpus\|ansible_mem"
```

Do you see these default ansible variables? For example I see:

```
# ansible -i /etc/ansible/hosts ocp311-m1.local -m setup | grep "ansible_processor_vcpus\|ansible_mem"
        "ansible_memfree_mb": 1036, 
        "ansible_memory_mb": {
        "ansible_memtotal_mb": 2870, 
        "ansible_processor_vcpus": 4,
```
